packages/client/screens/PasswordResetConfirmationTokenScreen.tsx

```diff
@@ -0,0 +1,221 @@
+import { useSelector } from '@xstate/react';
+import { Text, useSx, View } from 'dripsy';
+import React, { useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { TouchableOpacity } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import {
+    AppScreen,
+    AppScreenContainer,
+    AppScreenHeader,
+    TextField,
+} from '../components/kit';
+import { useAppContext } from '../contexts/AppContext';
+import { PasswordResetConfirmationTokenScreenProps } from '../types';
+
+interface PasswordResetConfirmationTokenFormFieldValues {
+    code: string;
+}
+
+const PasswordResetConfirmationTokenScreen: React.FC<PasswordResetConfirmationTokenScreenProps> =
+    ({ navigation }) => {
+        const insets = useSafeAreaInsets();
+        const sx = useSx();
+
+        const {
+            control,
+            handleSubmit,
+            setError,
+            formState: { errors },
+        } = useForm<PasswordResetConfirmationTokenFormFieldValues>({
+            defaultValues: {
+                code: '',
+            },
+        });
+
+        const { appService } = useAppContext();
+        const passwordResetCodeIsInvalid = useSelector(appService, (state) =>
+            state.hasTag('passwordResetCodeIsInvalid'),
+        );
+
+        useEffect(() => {
+            if (passwordResetCodeIsInvalid === false) {
+                return;
+            }
+
+            setError('code', {
+                type: 'server',
+                message: 'Code is invalid.',
+            });
+        }, [passwordResetCodeIsInvalid, setError]);
+
+        function handlePasswordResetTokenSubmit({
+            code,
+        }: PasswordResetConfirmationTokenFormFieldValues) {
+            appService.send({
+                type: 'SUBMIT_PASSWORD_RESET_CONFIRMATION_FORM',
```

Comment:
thoughts: I'm not sure about this event name, unless I'm mistaken this event will perform a reset password token verification and not a password reset confirmation ( that, in my mind, could be the next step after validating a password reset code where user will have to define a new password )